### PR TITLE
Properly updating scapegoat to 1.4.12

### PR DESCRIPTION
**Pull request by sebastian-peter**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -125,8 +126,8 @@ dependencies {
   implementation 'org.locationtech.jts.io:jts-io-common:1.18.2'
```
[the remark this answers is not in the document]
Comment (johanneshiry) [reply build.gradle:126]: @sebastian-peter would you mind excluding junit pls (cf. build.gradle @ PowerSystemUtils)

File: src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala [hunk 2]
```diff
@@ -1191,16 +1191,16 @@ trait DBFSAlgorithm extends PowerFlowSupport with GridResultsSupport {
...
-          currentTimestamp
+    gridAgentBaseData.sweepValueStores.lastOption match {
+      case Some((_, valueStore)) =>
+        notifyListener(
+          this.createResultModels(
+            gridAgentBaseData.gridEnv.gridModel,
+            valueStore
+          )(
+            currentTimestamp
+          )
```
Comment (johanneshiry) [inline src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala:1202]: Match is not exhaustive. 

Therefore I would propose sth like e.g.:

```
    gridAgentBaseData.sweepValueStores.lastOption.map {
      case (_, valueStore) =>
        notifyListener(
          this.createResultModels(
            gridAgentBaseData.gridEnv.gridModel,
            valueStore
          )(
            currentTimestamp
          )
```

File: src/main/scala/edu/ie3/simona/model/participant/load/profile/LoadProfileStore.scala [hunk 3]
```diff
@@ -215,15 +219,26 @@ object LoadProfileStore extends LazyLogging {
...
                   .getOrElse(0d)
               })
-              .max
+              .maxOption
+              .getOrElse(
+                // this should not happen - there's more than one DayType
+                throw new RuntimeException(
+                  "Maximum load not found."
+                )
+              )
```
Comment (johanneshiry) [inline src/main/scala/edu/ie3/simona/model/participant/load/profile/LoadProfileStore.scala:228]: If this cannot happen, avoid the exception. This means, please refactor this part such that a mapping is only produced if maxConsuption is available. Something like the following pseudo code should do the trick: 

```
knownLoadProfiles
      .flatMap(loadProfile => {
       loadProfile match {
         case BdewLoadProfile.H0 => 
            ....maxOption
        }.map(maxConsumption => loadProfile -> maxConsumption)
      }).toMap
```

File: src/main/scala/edu/ie3/simona/service/weather/WeatherSource.scala [hunk 4]
```diff
@@ -174,62 +174,63 @@ trait WeatherSource {
...
-            coordinateDistance.getCoordinateB -> weight
-          })
-          .toMap
-
-        val weightSum = weightMap.values.sum
-        if (weightSum > 0.99 && weightSum < 1.01)
-          Success(WeightedCoordinates(weightMap))
-        else
+      case _ =>
+        /* There is more than one coordinate existent */
```
Comment (johanneshiry) [inline src/main/scala/edu/ie3/simona/service/weather/WeatherSource.scala:184]: Invalid comment. There may be more than ONE or NONE coordinates existent

File: src/main/scala/edu/ie3/simona/sim/setup/SetupHelper.scala [hunk 5]
```diff
@@ -96,7 +96,7 @@ trait SetupHelper extends LazyLogging {
   ): Map[SubGridGate, ActorRef] =
     subGridGates
       .groupBy(gate => (gate.getSuperiorNode, gate.getInferiorNode))
-      .map(_._2.head)
+      .flatMap(_._2.headOption)
```
Comment (johanneshiry) [inline src/main/scala/edu/ie3/simona/sim/setup/SetupHelper.scala:99]: :+1:

File: src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala [hunk 6]
```diff
@@ -182,15 +182,13 @@ class ResultEventListenerSpec
         val resultFileLines = resultFileSource.getLines().toVector
 
         resultFileLines.size shouldBe 2
-        resultFileLines.tail.size shouldBe 1
```
Comment (johanneshiry) [inline src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala:185]: Why the removed equals to one element?

**Review by johanneshiry — CHANGES_REQUESTED** [review]
Thx for your work. Added two suggestions and one question. :)

**Review by johanneshiry — APPROVED** [review]
Perfect! Thank you! 🥇

**Review by ckittl — APPROVED** [review]

**Pull request merged**